pkg/client/unversioned/clientcmd/loader.go

```diff
@@ -328,11 +328,19 @@ func LoadFromFile(filename string) (*clientcmdapi.Config, error) {
 		return nil, err
 	}
 	glog.V(6).Infoln("Config loaded from file", filename)
-
+		
 	// set LocationOfOrigin on every Cluster, User, and Context
 	for key, obj := range config.AuthInfos {
 		obj.LocationOfOrigin = filename
+		if len(obj.TokenFile) > 0 {
+			tokenBytes, err := ioutil.ReadFile(obj.TokenFile)
```

Comment:
There ought to be some relativization logic somewhere to makes sure the filename is resolved relative to the kubeconfig location, not the current working directory.  Did you find it?